### PR TITLE
Create Nodegroups in an EKS cluster

### PR DESCRIPTION
**Pull request by mirshahriar**
#### Summary

```bash
$ cloud cluster nodegroups create --cluster=mw3xfmepaffefbuwees6ynbqjo --nodegroups calls=SizeAlefDev
```

This command will create a new Nodegroup in an existing EKS cluster.

#### Ticket Link

  Fixes https://mattermost.atlassian.net/browse/MM-51736

#### Release Note

```release-note
None
```

### CODE REVIEW

File: model/cluster_request.go [hunk 1]
```diff
@@ -379,3 +379,75 @@ func NewProvisionClusterRequestFromReader(reader io.Reader) (*ProvisionClusterRe
...
+	Nodegroups                 map[string]NodeGroupMetadata `json:"nodegroups"`
+	NodeGroupWithPublicSubnet  []string                     `json:"nodegroup-with-public-subnet,omitempty"`
+	NodeGroupWithSecurityGroup []string                     `json:"nodegroup-with-sg,omitempty"`
+}
+
+// SetDefaults sets default values for nodegroups.
+func (request *CreateNodegroupsRequest) SetDefaults() {
+	for ng, meta := range request.Nodegroups {
+		if len(meta.InstanceType) == 0 {
+			meta.InstanceType = "m5.large"
```
Comment (fmartingr) [inline model/cluster_request.go:393]: I keep seeing instance sizes around the code for defaults, shouldn't we put this in a default constant somewhere?

**Review by fmartingr — APPROVED** [review]
lgtm! just one minor comment

File: model/cluster_request.go (hunk 1 above)
Comment (mirshahriar) [reply model/cluster_request.go:393]: @gabrieljackson, we have some items like this. Do we want to make them const?

Comment (gabrieljackson) [reply model/cluster_request.go:393]: Yep, that's probably a good idea, but I am fine doing this as a follow-up.

**Review by gabrieljackson — APPROVED** [review]
LGTM

**Pull request merged**